File: ChatBot/HelperClasses/Logger.vb
Imports System.Runtime.InteropServices
Public Class Logger
    Dim strErrorFilePath As String
    Dim Stack As New ArrayList
    Public Sub New(Name As String, message As String)
        'Call Log Error
        strErrorFilePath = Environment.GetFolderPath(System.Environment.SpecialFolder.MyDocuments) & "\Silver Monkey\Log\" & Name & Date.Now().ToString("MM_dd_yyyy_H-mm-ss") & ".txt"
        System.IO.Directory.CreateDirectory(Environment.GetFolderPath(System.Environment.SpecialFolder.MyDocuments) & "\Silver Monkey\Log\")
        LogMessage(message)
    End Sub
    Public Function IsFileInUse(ByVal filePath As String) As Boolean
        Try
            Dim contents() As String = IO.File.ReadAllLines(filePath)
        Catch ex As IO.IOException
            Return (ex.Message.StartsWith("The process cannot access the file") AndAlso _
                    ex.Message.EndsWith("because it is being used by another process."))
        Catch ex As Exception
            Return False
        End Try
        Return False
    End Function

    Public Sub LogMessage(Message As String)

        Dim ioFile As System.IO.StreamWriter = Nothing
        Try

            ioFile = New System.IO.StreamWriter(strErrorFilePath, True)
            For Each line In Stack.ToArray
                ioFile.WriteLine(line)
            Next
            Stack.Clear()
            ioFile.WriteLine(Message)

            ioFile.Close()
        Catch ex As IO.IOException
            If (ex.Message.StartsWith("The process cannot access the file") AndAlso _
                    ex.Message.EndsWith("because it is being used by another process.")) Then
                Stack.Add(Message)
            End If

        Catch exLog As Exception
            If Not IsNothing(ioFile) Then
                ioFile.Close()
            End If
        End Try
    End Sub

   



End Class

File: Verbot5Library/Logger.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Runtime.InteropServices;

namespace Conversive.Verbot5
{

    public class Logger
    {
        string strErrorFilePath;
        ArrayList Stack = new ArrayList();
        public Logger(string Name, string message)
        {
            //Call Log Error
            strErrorFilePath = Environment.GetFolderPath(System.Environment.SpecialFolder.MyDocuments) + "\\Silver Monkey\\Log\\" + Name + System.DateTime.Now.ToString("MM_dd_yyyy_H-mm-ss") + ".txt";
            System.IO.Directory.CreateDirectory(Environment.GetFolderPath(System.Environment.SpecialFolder.MyDocuments) + "\\Silver Monkey\\Log\\");
            LogMessage(message);
        }
        public bool IsFileInUse(string filePath)
        {
            try
            {
                string[] contents = System.IO.File.ReadAllLines(filePath);
            }
            catch (System.IO.IOException ex)
            {
                return (ex.Message.StartsWith("The process cannot access the file") && ex.Message.EndsWith("because it is being used by another process."));
            }
            catch (Exception ex)
            {
                return false;
            }
            return false;
        }


        public void LogMessage(string Message)
        {
            System.IO.StreamWriter ioFile = null;

            try
            {
                ioFile = new System.IO.StreamWriter(strErrorFilePath, true);
                foreach (string line in Stack.ToArray())
                {
                    ioFile.WriteLine(line);
                }
                Stack.Clear();
                ioFile.WriteLine(Message);

                ioFile.Close();
            }
            catch (System.IO.IOException ex)
            {
                if ((ex.Message.StartsWith("The process cannot access the file") && ex.Message.EndsWith("because it is being used by another process.")))
                {
                    Stack.Add(Message);
                }

            }
            catch (Exception exLog)
            {
                if ((ioFile != null))
                {
                    ioFile.Close();
                }
            }
        }




    }
}
